jbmc/unit/java_bytecode/java_static_initializers/assignments_from_json.cpp

```diff
@@ -0,0 +1,412 @@
+    {
+      REQUIRE(
+        make_query(block)[2]
+          .as<code_assignt>()[0]
+          .as<symbol_exprt>()
+          .get()
+          .get_identifier() == "symbol_to_assign");
+    }
+    THEN("The instruction zero-initializes the struct")
+    {
+      REQUIRE(
+        make_query(block)[3].as<code_assignt>()[1].get().type() ==
+        test_class_type.subtype());
+    }
+    THEN("The instruction assigns the field to 42")
+    {
+      REQUIRE(
+        make_query(block)[4]
+          .as<code_assignt>()[0]
+          .as<member_exprt>()
+          .get()
+          .get_component_name() == "field_name");
+      REQUIRE(
+        numeric_cast_v<mp_integer>(make_query(block)[4]
+                                     .as<code_assignt>()[1]
+                                     .as<constant_exprt>()
+                                     .get()) == 42);
+    }
+  }
+
+  GIVEN("A class with an array field")
+  {
+    const json_objectt json = [] {
+      json_objectt entry{};
+      entry["array_field"] = [] {
+        json_objectt array_json;
+        array_json["@items"] = json_arrayt{json_numbert{"42"}};
+        array_json["@type"] = json_stringt{"[I"};
+        return array_json;
+      }();
+      entry["@type"] = json_stringt{"TestClass"};
+      return entry;
+    }();
+    class_to_declared_symbols.emplace("java::TestClass", [] {
+      symbolt field_symbol;
+      field_symbol.base_name = "array_field";
+      field_symbol.name = "field_name_for_codet";
+      field_symbol.type = java_int_type();
+      field_symbol.is_static_lifetime = true;
+      return field_symbol;
+    }());
+    const irep_idt clinit_name = add_clinit(symbol_table, "TestClass");
+    add_class_with_components(
+      symbol_table,
+      "TestClass",
+      {std::pair<std::string, typet>("@class_identifier", string_typet{}),
+       std::pair<std::string, typet>("array_field", java_array_type('i'))});
+    // For array[int]
+    add_java_array_types(symbol_table);
+
+    const reference_typet test_class_type =
+      reference_type(struct_tag_typet{"java::TestClass"});
+    code_blockt block;
+    optionalt<ci_lazy_methods_neededt> option{};
+    assign_from_json(
+      symbol_exprt{"symbol_to_assign", test_class_type},
+      json,
+      clinit_name,
+      block,
+      symbol_table,
+      option,
+      max_user_array_length,
+      references);
+
+    THEN("The instruction is the declaration of a symbol of TestClass* type")
+    {
+      const symbol_exprt allocation_symbol =
+        make_query(block)[0].as<code_declt>()[0].as<symbol_exprt>().get();
+      REQUIRE(allocation_symbol.type() == test_class_type);
+      THEN("The instruction declares the array data")
+      {
+        REQUIRE(
+          make_query(block)[1]
+            .as<code_declt>()[0]
+            .as<symbol_exprt>()
+            .get()
+            .type() == java_reference_type(java_int_type()));
+      }
+      THEN("The instruction allocates the struct")
```

Comment:
:question: Which struct? All this is checking for is an assignment to `allocation_symbol`.